src/lib.rs

```diff
@@ -427,9 +430,22 @@ impl Member {
         let repr_into = &self.repr_into;
         let repr_from = &self.repr_from;
         let bits = self.bits as u32;
-        quote!{
-            let mask = #base_ty::MAX >> (#base_ty::BITS - #bits);
-            this.0 = #repr_from(#repr_into(this.0) | (((#default as #base_ty) & mask) << #offset));
+        let order = self.order;
+
+        match order {
```

Comment:
I probably would this check this earlier in the constructor of Member and calculate the correct offset depending on the order there. Like I've done here: 

https://github.com/wrenger/bitfield-struct-rs/blob/main/src/lib.rs#L362-L366

The bug is that this calculation is only done for non-padding fields. So we can just move this before the if case.

Later, we can just use the offset without having to know the order.

Reply:
Applied.